plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -188,11 +188,179 @@ def __cmp__(self, other):
+    images/macboot.img = sha256:da76ff5490b4ae7e123f19b8f4b36efd6b7c435073551978d50c5181852a87f5
+    images/product.img = sha256:ffce14a7a95be20b36f302cb0698be8c19fda798807d3d63a491d6f7c1b23b5b
+    images/boot.iso = sha256:56af126a50c227d779a200b414f68ea7bcf58e21c8035500cd21ba164f85b9b4
+    images/pxeboot/vmlinuz = sha256:81c28a439f1d23786057d3b57db66e00b2b1a39b64d54de1a90cf2617e53...
+    images/pxeboot/initrd.img = sha256:aadebd07c4c0f19304f0df7535a8f4218e5141602f95adec08ad1e22f...
+    images/pxeboot/upgrade.img = sha256:224d098fb3903583b491692c5e0e1d20ea840d51f4da671ced97d422...
+    repodata/repomd.xml = sha256:3af1609aa27949bf1e02e9204a7d4da7efee470063dadbc3ea0be3ef7f1f4d14
+    ```
+
+    Note that not all fields may be present. For example, some RHEL 5 repositories do not
+    contain the `variant` field in their "treeinfo" files.
+
+    :ivar distribution_id: The concatenation of the values of `family`, `variant`, `version`,
+                           and `arch`
+    :ivar family:          Equivalent to the "treeinfo" [release] section's 'name' field. For
+                           example, 'Fedora', 'Red Hat Enterprise Linux', or 'Spacewalk'.
+    :ivar variant:         The variant contained in the repository. The current version of the
+                           model is problematic since some old "treeinfo" files don't have
+                           variants, and some new repositories have _several_ variants. Since
+                           we currently use the [general] section of the "treeinfo" file, this
+                           variant will be the first variant (alphabetically) in the distribution
+                           tree.
+    :ivar version:         Equivalent to the "treeinfo" [release] section's 'version' field.
+                           For example, '21', '7.0', and '2.1'
+    :ivar arch:            Equivalent to the "treeinfo" [tree] section's 'arch' field. The
+                           architecture the tree was built for. For example, 'x86_64'.
+    :ivar files:           A somewhat hodgepodge list of files in the distribution. It contains
+                           some of the files described in the "treeinfo" files, and potentially
+                           files described by an XML file called "PULP_DISTRIBUTION.xml".
+    :ivar timestamp:       Equivalent to the "treeinfo" [tree] section's 'build_timestamp' field.
+    :ivar packagedir:      Equivalent to the "treeinfo" [variant-*] section's 'packages' field.
+                           It's possible in the future that distribution trees will have several
+                           different packagedir fields, so we need to fix our model here.
+    :ivar version_sort_index: ?
+    """
 
     distribution_id = mongoengine.StringField(required=True)
     family = mongoengine.StringField(required=True)
-    variant = mongoengine.StringField(required=True)
+    variant = mongoengine.StringField()
```

Comment:
A quick grep leads me to think a Distribution is only created in one place: https://github.com/pulp/pulp_rpm/blob/master/plugins/pulp_rpm/plugins/importers/yum/parse/treeinfo.py#L500-L521

That query for an existing unit will work on old units with ``variant: null`` (assuming the value of the ``variant`` kwarg is ``None``), so a second unit won't be created.

As an aside, since releases are supposed to be immutable[0] and a distribution tree describes a subset of a release, modifying the unit the way the above code does seems bad, but that's a different problem. 

[0] http://release-engineering.github.io/productmd/terminology.html#release